### PR TITLE
Fix npm install resolution of peer deps

### PR DESCRIPTION
**Pull request by jonchurch**
# The Issue

Since #19 introduced canary versions of core libraries, dep installation fails when running `npm i` with npm version 7 and later (Node 15 and up). 

I'm not actually sure what is failing here, as the ranges in the below error message should be satisfied by the version found. Semver and prereleases are a little counterintuitive, but boil down to any pre-release being seen as lower than the version it is suffixed to. 

E.g:
* 1.0.0-alpha does not satisfy ^1.0.0, but 1.0.0 does.
* 1.0.0-alpha satisfies ^1.0.0-alpha.
* Crucially, 1.0.1-alpha should satisfy ^1.0.0 (Slightly below 1.0.1 is still larger than 1.0.0)

> **Specifically in this case it's the canary release of next, it doesn't satisfy the peerdep requirements of of the experimental next-auth package.**
```
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR!
npm ERR! While resolving: next-template@0.0.2
npm ERR! Found: next@13.4.7-canary.1
npm ERR! node_modules/next
npm ERR!   next@"13.4.7-canary.1" from the root project
npm ERR!
npm ERR! Could not resolve dependency:
npm ERR! peer next@"^13.4.6" from next-auth@0.0.0-manual.83c4ebd1
npm ERR! node_modules/next-auth
npm ERR!   next-auth@"0.0.0-manual.83c4ebd1" from the root project
npm ERR!
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
```
Apparenly pnpm's peer dep resolution behaves differently, which is why you don't get this error when using pnpm. Although I'm not sure in what way it's different.

## Motivation

This issue impacts users on recent Node versions when they attempt to run the standard npm i command on this starter template. I know yall are using `pnpm`. But recognizing that the primary audience might be using the template for their initial foray into the Node ecosystem, I'd encourage you to offer a smoother experience. The error message, as it currently stands, is confusing for many.

Yes folks can do force install, or install with old peer dep behavior. But again, unless you have a good compelling reason that this starter needs to be on canaries, let's not burden the user.

# Proposed Change

This PR suggests transitioning away from the canary versions of next and eslint-config-next. The versions will now adhere to a caret range, initialized at the @latest version as of this PR (^v13.4.12).

### CODE REVIEW

**Comment by jonchurch** [comment]
This seems related to https://github.com/npm/cli/issues/4958 for anyone curious. It's not the exact same issue, as the linked issue is failing to satisfy a range version that looks like `>= 10.2.0` whereas this issue is a more common caret range.

TLDR it's an npm feature/bug.

**Comment by leerob** [comment]
Closing since we'll want to update all dependencies together
